javascript/javascript2/week2/main2.js

```diff
@@ -0,0 +1,31 @@
+console.log("Script loaded");
+
+function renderProducts(products) {
+  const ul = document.createElement('section ul');
+
+  products.forEach(product => { 
+    const li = document.createElement('li'); 
+    li.innerHTML = `
+      <h3>${product.name}</h3>
+      <p>Price: ${product.price}</p>
+      <p>Rating: ${product.rating}</p>
+    `;
+    ul.appendChild(li); 
+  });
+
+  document.body.appendChild(ul); 
+}
+
+const products = getAvailableProducts();
+renderProducts(products); 
+
+const searchInput = document.getElementById('search-input');
+searchInput.addEventListener('input', () => {
```

Comment:
Instead of writing the callback function here, try to write it as a seperate function and call that from here

Reply:
Ok, Thank you so much for your feedback. I fix it now.